ssl/statem/extensions_srvr.c

```diff
@@ -89,7 +89,7 @@ int tls_parse_ctos_server_name(SSL *s, PACKET *pkt, unsigned int context,
     /*
      * Although the server_name extension was intended to be
      * extensible to new name types, RFC 4366 defined the
-     * syntax inextensibly and OpenSSL 1.0.x parses it as
+     * syntax extensibility and OpenSSL 1.0.x parses it as
```

Comment:
you can replace "a hostname" with "exactly one hostname" if you prefer.  Actually since it's your PR, you can replace it with anything you want :)